src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -101,6 +99,29 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     }
                 }
             }
+
+            await RemoveSecondaryIndex(sagaData, table).ConfigureAwait(false);
```

Comment:
Too bad main record and the secondary index records don't share the same PartitionKey. If they'd do, batched deletion would be atomic.